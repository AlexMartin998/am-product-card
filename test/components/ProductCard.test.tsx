import * as React from 'react';
import { create, act } from 'react-test-renderer';

import { testProduct2 } from '../data/product';
import { ProductCard } from '../../src/components/index';

describe('first', () => {
  test('should return a correctly rendered component', () => {
    const wrapper = create(
      <ProductCard product={testProduct2} className="custom-class-product-card">
        {() => <h1>Hello Test!</h1>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should increase counter', () => {
    const wrapper = create(
      <ProductCard product={testProduct2} className="custom-class-product-card">
        {({ count, increaseBy }) => (
          <>
            <h1>Hello Test - Counter!</h1>

            <span>{count}</span>
            <button onClick={() => increaseBy(1)}>+1</button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    });
    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('1');
  });
});
