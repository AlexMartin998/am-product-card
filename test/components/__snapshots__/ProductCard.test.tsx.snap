// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first should increase counter 1`] = `
<div
  className="productCard custom-class-product-card"
>
  <h1>
    Hello Test - Counter!
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;

exports[`first should return a correctly rendered component 1`] = `
<div
  className="productCard custom-class-product-card"
>
  <h1>
    Hello Test!
  </h1>
</div>
`;
